scripts/publish.js

```diff
@@ -0,0 +1,37 @@
+const { argv } = require('process');
+const utils = require('util');
+const exec = utils.promisify(require('child_process').exec);
+
+async function publish(publishType) {
+  if (!publishType || !publishType.match(/major|minor|patch/)) {
+    console.error(
+      'Invalid publish type: Available types are major|minor|patch'
+    );
+    process.exit(1);
+  }
+
+  try {
+    /* 1.  Make sure the bump type is correct */
+    await exec(`npm version ${publishType} --git-tag-version=false`);
+
+    /* 2.  Build and test with the correct version. Building the library will also update the src/info.ts file */
+    await exec('yarn build');
+    await exec('yarn test');
+
+    /* 3.  Git actions */
+    await exec('git add -A');
+    await exec(
+      `git commit -am "Bump release ${require('../package.json').version}"`
+    );
+    await exec(`git tag -a v${require('../package.json').version}`);
```

Comment:
🙃 You can require package.json once on top.

Reply:
@SokratisVidros Actually this cannot be done in Node and it will bring the opposite behaviour of what we want for two reasons: _(Putting this here for future reference also)_
1. `require` is synchronous
2. `require` is keeping an internal cache _(making it use the 'previous' package version before the bump)_

The `npm version` command bumps the package.json version and then we need to make sure we pick the latest one. What we could do is creating a `const` after the `npm version` command but I would not prefer to make it easier for someone to just change the order of definition and break the script.